modules/js-and-css/speculation-rules/readme.txt

```diff
@@ -0,0 +1,84 @@
+=== Speculation Rules ===
+
+Contributors:      wordpressdotorg
+Requires at least: 6.3
+Tested up to:      6.4
+Requires PHP:      7.0
+Stable tag:        1.0.0
+License:           GPLv2 or later
+License URI:       https://www.gnu.org/licenses/gpl-2.0.html
+Tags:              performance, javascript, speculation rules, prerender, prefetch
+
+Uses the Speculation Rules API to prerender linked URLs upon hover.
+
+== Description ==
+
+This plugin adds support for the [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/API/Speculation_Rules_API), which allows defining rules by which certain URLs are dynamically prefetched or prerendered based on user interaction.
+
+[See the Speculation Rules WICG specification draft.](https://wicg.github.io/nav-speculation/speculation-rules.html)
+
+By default, the plugin is configured to prerender WordPress frontend URLs when the user hovers over a relevant link. This can be customized via the "Speculation Rules" section under _Settings > Reading_.
+
+A filter can be used to exclude certain URL paths from being eligible for prefetching and prerendering (see FAQ section). Alternatively, you can add the 'no-prerender' CSS class to any link (`<a>` tag) that should not be prerendered.
+
+= Browser support =
+
+The Speculation Rules API is a new web API and currently requires using Chrome 121+.
```

Comment:
```suggestion
The Speculation Rules API is a relatively new web API and currently requires using Chrome 121+ (or equivalent Chromium-based browser) for the features of this that we are using.
```

The Speculation Rules API was available well before 121 but not for this feature. So suggest this to make it clearer.

Also it will be available in other browsers (not sure if from 121 version of those - @jeremyroman when you say "enabled by default from 121" do you mean no longer in Finch or still controlled by Finch so not yet available in other Chromium browsers based off of 121?

Reply:
While the API was available before, the relevant functionality of it which this plugin is based upon (document rules) are only available in Chrome 121+ as far as I'm aware (before they required origin trial, so not really live). So while older versions still have the API, they wouldn't be able to use this plugin.

Reply:
The functionality mentioned here is out of origin trial and enabled by default (not only by Finch) in Chrome 121. And the code substantially lives in content/.

So other Chromium-based browsers would get it unless they disabled it. For instance, it works in Microsoft Edge 121 (beta) right now (just checked on my Windows VM), so unless Microsoft disables it between now and when that ships to stable, it will be available in Edge as well.

Some tweaks to the heuristics might ship outside Chromium proper and so be Google Chrome-specific, but not the availability of the API itself.

Reply:
Good to know. Whether you wanna mention this is up to you @felixarntz or if "Chrome 121" is good enough for your user base (and those that know will know that also means things like Edge soon enough after).

Reply:
Maybe we go with just "Chrome 121" for now (i.e. keep this as is) and modify if we're certain Edge support is present when that version launches? That'll most likely still be before this plugin is published anyway, so we can just fix it in another PR.